packages/application/src/shell.ts

```diff
@@ -1551,6 +1560,82 @@ namespace Private {
     private _lastCurrent: Widget | null;
   }
 
+  export class SkipLinkWidgetHandler {
+    /**
+     * Construct a new skipLink widget handler.
+     */
+    constructor(shell: ILabShell) {
+      const skipLinkWidget = (this._skipLinkWidget = new Widget());
+      const skipToFilterFiles = document.createElement('a');
+      skipToFilterFiles.href = '#';
+      skipToFilterFiles.tabIndex = 1;
+      skipToFilterFiles.text = 'Skip to Filter Files';
+      skipToFilterFiles.className = 'skip-link';
+      skipToFilterFiles.addEventListener('click', this);
+      skipLinkWidget.addClass('jp-skiplink');
+      skipLinkWidget.id = 'jp-skiplink';
+      skipLinkWidget.node.appendChild(skipToFilterFiles);
+    }
+
+    handleEvent(event: Event): void {
+      switch (event.type) {
+        case 'click':
+          this._focusFileSearch();
+          break;
+      }
+    }
+
+    private _focusFileSearch() {
+      const input = document.querySelector(
+        '#filename-searcher .bp3-input'
```

Comment:
```suggestion
        '#filename-searcher input'
```

`.bp3-input` is a styling class that may change in the future. Searching for just the `input` tag will make this less fragile